cmd/agent/api/grpc.go

```diff
@@ -185,6 +187,64 @@ func (s *serverSecure) TaggerFetchEntity(ctx context.Context, in *pb.FetchEntity
 	}, nil
 }
 
+func (s *serverSecure) GetConfigs(ctx context.Context, in *pb.GetConfigsRequest) (*pb.GetConfigsResponse, error) {
+	if s.configService == nil {
+		log.Debug("Remote configuration service not initialized")
+		return nil, errors.New("remote configuration service not initialized")
+	}
+
+	configs, err := s.configService.GetConfigs(in.Product.String())
+	if err != nil {
+		return nil, err
+	}
+
+	return &pb.GetConfigsResponse{
+		ConfigResponses: configs,
+	}, nil
+}
+
+func (s *serverSecure) GetConfigUpdates(in *pb.SubscribeConfigRequest, out pb.AgentSecure_GetConfigUpdatesServer) error {
+	if s.configService == nil {
+		log.Debug("Remote configuration service not initialized")
+		return errors.New("remote config service not initialized")
+	}
+
+	ctx := out.Context()
+	configs := make(chan *pb.ConfigResponse, 10)
+
+	log.Debugf("New remote configuration subscriber request for product %s", in.Product)
+	subscriber := remoteconfig.NewSubscriber(in.Product, time.Second, func(config *pb.ConfigResponse) error {
+		log.Debug("Pushing configuration for gRPC client")
+		select {
+		case configs <- config:
+			log.Debug("Pushed configuration to gRPC client")
+			return nil
+		default:
+			return errors.New("failed to notify gRPC subscriber")
+		}
+	})
+
+	log.Debugf("New remote configuration subscriber for product %s", in.Product)
+	s.configService.RegisterSubscriber(subscriber)
+
+LOOP:
+	for {
+		log.Debug("Streaming config to gRPC client")
+		select {
+		case config := <-configs:
+			log.Debugf("Sending configuration for product %s", in.Product)
+			if err := out.Send(config); err != nil {
```

Comment:
If the client closed the connection, it will then go through `case <-ctx.Done():` and unregister the client. But it's true we can probably exit directly'